### PR TITLE
Fix test cleanup of RunParallel

### PR DESCRIPTION
**Pull request by howardjohn**
I am guessing this will solve
https://github.com/istio/istio/issues/24028

I am not proud of this solution, but I spent quite a few hours on this
and its the best I could come up with. The sequencing here is quite
complicated...

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by stevenctl** [comment]
Curious if this is related to https://github.com/istio/istio/issues/23447. I found if I wrapped `RunParallel` with a `NewSubTest` it fixed things.

File: pkg/test/framework/test.go [hunk 1]
```diff
@@ -204,8 +208,7 @@ func (t *Test) runInternal(fn func(ctx TestContext), parallel bool) {
 	}
 	suiteName := t.s.settings.TestID
 	if len(t.featureLabels) < 1 && !features.GlobalWhitelist.Contains(suiteName, myGoTest.Name()) {
-
-		myGoTest.Fatalf("Detected new test %s in suite %s with no feature labels.  "+
+		myGoTest.Logf("Detected new test %s in suite %s with no feature labels.  "+
```
Comment (therealmitchconnors) [inline pkg/test/framework/test.go:211]: What is the reason for this change?  From docs, it looks like this will not fail the related tests, which is the desired outcome..

Comment (howardjohn) [reply pkg/test/framework/test.go:211]: This wasn't actually intended to be merged, but I got frustrated because my options are to either disable this, merge https://github.com/istio/istio/pull/24875, add some bogus feature, or spend an hour adding whitelists.... its a horrible dev UX without the linked PR

File: pkg/test/framework/integration/framework_test.go [hunk 2]
```diff
@@ -90,3 +93,49 @@ func TestParallel(t *testing.T) {
 	assertClosedBefore(t, l1a, top)
 	assertClosedBefore(t, l1b, top)
 }
+
+// Validate that cleanup is done synchronously for asynchronous tests
+func TestParallelWhenDone(t *testing.T) {
+	errors := make(chan error, 10)
+	framework.NewTest(t).
```
Comment (therealmitchconnors) [inline pkg/test/framework/integration/framework_test.go:100]:
```suggestion
	framework.NewTest(t).Features("TestFramework").
```

Comment (therealmitchconnors) [reply pkg/test/framework/integration/framework_test.go:100]: This coupled with a one line change to features.yaml should make the test pass the whitelist.  Let me know if that is not the case.

File: pkg/test/framework/test.go (hunk 1 above)
Comment (howardjohn) [reply pkg/test/framework/test.go:211]: I have reverted this

**Comment by howardjohn** [comment]
I added the feature label
